lib/pluginmanager/install.rb

```diff
@@ -43,9 +43,11 @@ def validate_cli_options!
   # Check if the specified gems contains
   # the logstash `metadata`
   def verify_remote!(gems)
+    Gem.sources = Gem::SourceList.from(gemfile.gemset.sources)
```

Comment:
I think `Gem.sources` should be setup in `LogStash::PluginManager.logstash_plugin?`  right before the `Gem::SpecFetcher.fetcher.spec_for_dependency` call https://github.com/elastic/logstash/blob/0cf1a98b4a77f3278e814e1a1cb308fd184934ba/lib/pluginmanager/util.rb#L20  no?